### ISSUE TITLE
TRRTrajectoryFile offsets wrong

### ISSUE THREAD

**Issue by sefalkner** [mdtraj/mdtraj#1533]
Hello,

when reading a frame from a large gromacs 2018.6 TRR trajectory (10.000 frames, 60.000 particles, 14 GB) using the TRRTrajectoryFile reader as shown below, the offsets seem to be incorrect. I am using the latest conda release of mdtraj.

```
from mdtraj.formats import TRRTrajectoryFile

f = TRRTrajectoryFile("trajectory.trr", 'r')
try:
    f.seek(offset=4000)
    data = f._read(n_frames=1, atom_indices=None, get_velocities=True)
finally:
    trr.close()
```

Looking at the offsets, they continuously wrap from 2.1E9  to -2.1E9, in my case starting at frame 1590 (see plot). Although the dtype of the offset array is int64, these values suggest a conversion to int32 at some point I guess.

![offsets](https://user-images.githubusercontent.com/61740466/75777716-beed1300-5d56-11ea-8a5e-f2d2abeedf7f.png)

Reading from frame 1589 n_frames onward is no problem. It only occurs when jumping to a later frame with f.seek.

As libraries like OpenPathSampling heavily depend on reading just one frame at a given position, I would greatly appreciate some help on this issue.

**Comment by rmcgibbo** [comment]
The plot is the result of _calc_len_and_offsets?

**Comment by rmcgibbo** [comment]
Are you on a 32 or 64-bit OS?

**Comment by rmcgibbo** [comment]
And are you familiar with how to edit the code and compile mdtraj from source? It's probably going to be easier for me to give you some ideas and you test them out than for me to fix this, since I don't have a file like that hanging around.

**Comment by rmcgibbo** [comment]
I think you've diagnosed the problem correctly. e.g. try editing the "int" references in this function to uint64_t and recompiling. https://github.com/mdtraj/mdtraj/blob/master/mdtraj/formats/xtc/trr.pyx#L759

**Comment by sefalkner** [comment]
Sorry for missing out on some information. I'm on a 64 bit system and now compiled mdtraj from the current master branch. The plot was basically from _calc_len_and_offsets:

`plt.plot( f.offsets ) `

I think I found the problem. In https://github.com/mdtraj/mdtraj/blob/master/mdtraj/formats/xtc/trr.pyx#L759 the variable `frame_offset` is declared as int32. Later it temporarily stores the output of `xdrlib.xdr_tell(self.fh)`, which is int64. That's the cause for the strange wrapping at a value of 2.1E9. The numpy array `offsets` storing the offsets is correctly declared as int64.

I declared frame_offset as int64_t and the offsets are looking fine:

![offsets_corrected](https://user-images.githubusercontent.com/61740466/75869526-f9fc4e80-5e09-11ea-99db-77fd210c9fa1.png)

Seeking and reading is now possible at any point in the 10.000 frames, trying to read at 10.001 will give the expected TRR Seek out of bounds error.

Changing everything to uint64 requires some more changes in `xdrlib.xdr_tell`. I guess at first I'll submit a pull request with the small correction.
